Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/PropertyEditorNodes.h

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzFramework/DocumentPropertyEditor/DocumentSchema.h>
+
+namespace AZ::DocumentPropertyEditor::Nodes
+{
+    //! Adapter: The top-level tag for a DocumentAdapter that may contain any number of Rows.
+    struct Adapter : NodeDefinition
+    {
+        static constexpr AZStd::string_view Name = "Adapter";
+        static bool CanAddToParentNode(const Dom::Value& parentNode);
+        static bool CanBeParentToValue(const Dom::Value& value);
```

Comment:
repeated: these should be marked as overidden?

Reply:
These are static methods, so there's no polymorphism here - we're just relying on templating to expand to `TNodeDefinition::CanAddToParentNode` etc